src/data/chats/mod.rs

```diff
@@ -67,15 +69,6 @@ impl Chats {
     pub fn load_model(&mut self, file: &File, override_port: Option<u16>) {
         self.cancel_chat_streaming();
 
-        if let Some(mut chat) = self.get_current_chat().map(|c| c.borrow_mut()) {
```

Comment:
Not every time we load a model we mean to save it as the associated one. When using `@` to invoke a model, it will not be saved as an associated one. 
So this decision must happen outside of this method.